rotkehlchen/chain/ethereum/modules/scroll_bridge/decoder.py

```diff
@@ -0,0 +1,369 @@
+DEPOSIT_ERC20: Final = b'1\xcd;\x97nMe@"\xbf\x95\xc6\x8a,\xe5?\x1d]\x94\xaf\xab\xe0EM(2 \x8e\xeb@\xaf%'  # noqa: E501
+# FinalizeWithdrawERC20 - 0xc6f985873b37805705f6bce756dce3d1ff4b603e298d506288cce499926846a7
+FINALIZE_WITHDRAW_ERC20: Final = b'\xc6\xf9\x85\x87;7\x80W\x05\xf6\xbc\xe7V\xdc\xe3\xd1\xffK`>)\x8dPb\x88\xcc\xe4\x99\x92hF\xa7'  # noqa: E501
+
+
+class ScrollBridgeDecoder(DecoderInterface):
+    def __init__(
+            self,
+            evm_inquirer: 'EthereumInquirer',
+            base_tools: 'BaseDecoderTools',
+            msg_aggregator: 'MessagesAggregator',
+    ) -> None:
+        super().__init__(
+            evm_inquirer=evm_inquirer,
+            base_tools=base_tools,
+            msg_aggregator=msg_aggregator,
+        )
+        self.eth = A_ETH.resolve_to_crypto_asset()
+
+    def _decode_eth_deposit_withdraw(self, context: DecoderContext) -> DecodingOutput:
+        """Decodes ETH deposit and withdraw events. (Bridging ETH from and to ethereum)"""
+        if context.tx_log.topics[0] == DEPOSIT_ETH:
+            raw_amount = decode_abi(['uint256'], context.tx_log.data)[0]
+            user_address = hex_or_bytes_to_address(context.tx_log.topics[1])
+            amount = from_wei(FVal(raw_amount))
+            expected_event_type = HistoryEventType.SPEND
+            new_event_type = HistoryEventType.DEPOSIT
+            from_chain, to_chain = ChainID.ETHEREUM, ChainID.SCROLL
+        elif context.tx_log.topics[0] == FINALIZE_WITHDRAW_ETH:
+            raw_amount = decode_abi(['uint256'], context.tx_log.data[:32])[0]
+            user_address = hex_or_bytes_to_address(context.tx_log.topics[2])
+            amount = from_wei(FVal(raw_amount))
+            expected_event_type = HistoryEventType.RECEIVE
+            new_event_type = HistoryEventType.WITHDRAWAL
+            from_chain, to_chain = ChainID.SCROLL, ChainID.ETHEREUM
+        else:
+            return DEFAULT_DECODING_OUTPUT
+
+        if not self.base.is_tracked(user_address):
+            return DEFAULT_DECODING_OUTPUT
+
+        # Find the corresponding transfer event and update it
+        orig_amount = FVal(0)
+        bridge_event = self._get_eth_bridge_event(
+            context=context,
+            expected_event_type=expected_event_type,
+            sender=user_address,
+        )
+
+        if bridge_event is None:
+            return DEFAULT_DECODING_OUTPUT
+
+        orig_amount = bridge_event.balance.amount
+
+        self._edit_eth_bridge_event(
+            event=bridge_event,
+            new_event_type=new_event_type,
+            amount=amount,
+            from_chain=from_chain,
+            to_chain=to_chain,
+        )
+
+        refund_event_idx, refund_amount = self._get_refund_event_index_and_amount(
+            decoded_events=context.decoded_events,
+            sender=user_address,
+        )
+
+        if refund_event_idx is None:
+            return DEFAULT_DECODING_OUTPUT
```

Comment:
I don't know if this can happen in a real scenario... I will update it so that I handle that scenario. Basically we need to replace the refund event with the fees event that we create, so in case the refund event won't be found (refund is 0), we can just add the fee event.